### PR TITLE
drivers: dma: intel_lpss: Enhance LPSS init interface.

### PR DESCRIPTION
**Pull request by akanisetti**
Enhance LPSS DMA init to support both UART and I2C without dependency on parent node in dts.

Signed-off-by: Anisetti Avinash Krishna [anisetti.avinash.krishna@intel.com](mailto:anisetti.avinash.krishna@intel.com)

### CODE REVIEW

**Review by teburd — DISMISSED** [review]

**Comment by jhedberg** [comment]
@akanisetti looks good, but you really need to include the peripheral driver changes into this PR, since otherwise we end up in an intermediate state where DMA is broken.

File: dts/x86/intel/raptor_lake.dtsi [hunk 1]
```diff
@@ -237,7 +237,7 @@
 			clock-frequency = <100000000>;
 			interrupts = <PCIE_IRQ_DETECT IRQ_TYPE_LOWEST_LEVEL_LOW 3>;
 			interrupt-parent = <&intc>;
-			status = "okay";
+			status = "disabled";
```
Comment (finikorg) [inline dts/x86/intel/raptor_lake.dtsi:240]: why do you touch SPI in this commit?

**Comment by akanisetti** [comment]
> @akanisetti looks good, but you really need to include the peripheral driver changes into this PR, since otherwise we end up in an intermediate state where DMA is broken.

Updated

File: dts/x86/intel/raptor_lake.dtsi (hunk 1 above)
Comment (akanisetti) [reply dts/x86/intel/raptor_lake.dtsi]: Updated

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Pull request merged**